resources/ManageAPICredentials/ManageAPICredentialsActions.php

```diff
@@ -191,45 +203,42 @@ private function compareCredIds(APIAuthentication $cred1, APIAuthentication $cre
  * @param \APIAuthentication $api       Credential to warn about
  * @param int       $elapsedMonths      The number of months of non-use so far.
  * @param int       $deleteionThreshold The number of months of no-use which will trigger deletion if reached.
+ * @param bool      $isRenewalRequest   Flag indicating the presence or absence
+ *                                      of the `r` command-line argument.
  * @return void
  */
     private function sendWarningEmail(
         $fromEmail,
         $replyToEmail,
         \APIAuthentication $api,
         $elapsedMonths,
-        $deletionThreshold
+        $deletionThreshold,
+        $isRenewalRequest
     ) {
-        $user = $api->getUser();
-        $userEmail = $user->getEmail();
-        $siteName = $api->getParentSite()->getShortName();
-        $siteEmail = $siteName . ' <' . $api->getParentSite()->getEmail() . '>';
-
-        $headersArray = array ("From: $fromEmail",
-                           "Cc: $siteEmail");
-        if (strlen($replyToEmail) > 0 && $fromEmail !== $replyToEmail) {
-            $headersArray[] = "Reply-To: $replyToEmail";
-        }
-        $headers = join("\r\n", $headersArray);
-
         $subject = "GOCDB: Site API credential deletion notice";
 
-        $body = "Dear " . $user->getForename() . ",\n\n" .
-        "The API credential associated with the following identifier registered\n" .
-        "at site $siteName has not been used during\n" .
-        "the last $elapsedMonths months and will be deleted if this period of inactivity\n" .
-        "reaches $deletionThreshold months.\n\n";
+        list($headers, $siteName) = $this->getHeaderContent($api, $fromEmail, $replyToEmail);
+        list($body, $userEmail) = $this->getInitialBodyContent($api);
```

Comment:
With the body being split up like this, I find it hard to get a sense of what is being sent to the user.

Maybe a mapping between properties and body text would be easier alleviate this?

Reply:
Wrapped into single function that would provide fix to ease complexity.